packages/vite/src/node/plugins/css.ts

```diff
@@ -1387,7 +1384,13 @@ async function compileCSS(
     )
   }
 
-  if (urlReplacer) {
+  if (
+    urlReplacer &&
+    // if there's an @import, we need to add this plugin
+    // regradless of whether it contains url() or image-set(),
+    // because we don't know the content referenced by @import
+    (needInlineImport || cssUrlRE.test(code) || cssImageSetRE.test(code))
```

Comment:
```suggestion
    (needInlineImport || hasUrl)
```

It seems like we can reuse the variable here